### PR TITLE
Add get user api call and update types for error event support

### PR DESCRIPTION
**Pull request by dynbit**

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -188,3 +231,7 @@ type ResponsePayload struct {
 	ExtraMessages []*Message    `json:"extraMessages,omitempty"`
 	Conversation  *Conversation `json:"conversation,omitempty"`
 }
+
+type GetUserResponse struct {
```
Comment (pranas) [inline types.go]: `GetAppUserResponse` would be more consistent with the rest

File: smooch.go [hunk 2]
```diff
@@ -182,32 +183,61 @@ func (sc *smoochClient) Send(userID string, message *Message) (*ResponsePayload,
...
+	var responsePayload ResponsePayload
+	err = sc.sendRequest(http.MethodPost, url, buf, &responsePayload)
 	if err != nil {
 		return nil, err
 	}
+
+	return &responsePayload, nil
+}
+
+func (sc *smoochClient) VerifyRequest(r *http.Request) bool {
```
Comment (pranas) [inline smooch.go:154]: seems like this was just moved around and not a real change

File: types.go [hunk 3]
```diff
@@ -58,26 +59,68 @@ type ActionType string
...
 }
 
-type Source struct {
-	Type string `json:"type,omitempty"`
+type SourceDestination struct {
+	Type              string `json:"type,omitempty"`
+	Id                string `json:"id,omitempty"`
+	IntegrationId     string `json:"integrationId,omitempty"`
+	OriginalMessageId string `json:"originalMessageId,omitempty"`
+	// originalMessageTimestamp is skipped for now
```
Comment (pranas) [inline types.go]: not very useful

**Review by pranas — APPROVED** [review]

File: smooch.go (hunk 2 above)
Comment (dynbit) [reply smooch.go:154]: It is the way the this change get detected. Moved all unexported functions to the bottom.

**Pull request merged**